From c8541bbaa4f584b36614f0afa2a29bf3695c3ae9 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Tue, 31 Jul 2012 14:00:27 +0300
Subject: [PATCH 272/370] confc_invariant(): don't use invariants of RPC layer

Usage of c2_rpc_{session,conn}_invariant() by confc_invariant()
crosses module boundaries.

LogD #600
RB: r/939/diff/1/?file=25263#file25263line350
---
 conf/confc.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/conf/confc.c b/conf/confc.c
index b3d7f24..c2d99b5 100644
--- a/conf/confc.c
+++ b/conf/confc.c
@@ -343,10 +343,7 @@ static bool confc_invariant(const struct c2_confc *confc)
 	struct c2_rpc_client_ctx *rpc = &confc->cc_rpc;
 
 	return c2_confc_bob_check(confc) && confc->cc_root != NULL &&
-		confc->cc_group != NULL &&
-		ergo(rpc->rcx_remote_addr != NULL,
-		     c2_rpc_session_invariant(&rpc->rcx_session) &&
-		     c2_rpc_conn_invariant(&rpc->rcx_connection));
+		confc->cc_group != NULL;
 }
 
 /* ------------------------------------------------------------------
-- 
1.8.3.2

